spatialos-sdk/src/worker/op.rs

```diff
@@ -16,40 +16,81 @@ use crate::worker::internal::utils::*;
 use spatialos_sdk_sys::worker::*;
 
 pub struct OpList {
-    ops: Vec<WorkerOp>,
-    internal_ptr: *mut Worker_OpList,
+    raw: *mut Worker_OpList,
 }
 
 impl OpList {
-    pub(crate) fn new(raw_ops_list_ptr: *mut Worker_OpList) -> Self {
-        unsafe {
-            let worker_op_list = *raw_ops_list_ptr;
-            let ops = slice::from_raw_parts(worker_op_list.ops, worker_op_list.op_count as usize)
-                .iter()
-                .map(WorkerOp::from)
-                .collect::<Vec<WorkerOp>>();
-
-            OpList {
-                ops,
-                internal_ptr: raw_ops_list_ptr,
-            }
-        }
+    pub(crate) fn new(raw: *mut Worker_OpList) -> Self {
+        assert!(!raw.is_null());
+        OpList { raw }
+    }
+
+    /// Returns the number of ops in the list.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use spatialos_sdk::worker::connection::*;
+    /// # let connection: WorkerConnection = unimplemented!();
+    /// let op_list = connection.get_op_list(0);
+    /// println!("Number of ops received: {}", op_list.len());
+    /// ```
+    pub fn len(&self) -> usize {
+        self.raw().op_count as usize
+    }
+
+    /// Returns an iterator over the list.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use spatialos_sdk::worker::connection::*;
+    /// # let connection: WorkerConnection = unimplemented!();
+    /// for op in connection.get_op_list(0).iter() {
+    ///     // Process `op`.
+    /// }
+    /// ```
+    pub fn iter(&self) -> Iter<'_> {
```

Comment:
qq - What does `Iter<'_>` mean

Reply:
It's new in the 2018 edition. If I recall correctly, you have to specify a lifetime when referencing a type that has a lifetime parameter, i.e. I couldn't do `fn iter(&self) -> Iter`, I'd have to do `fn iter<'a>(&'a self) -> Iter<'a>`. Doing `'_` says "the value I'm returning borrows some data, but I'm letting the compiler infer the lifetime of that data". The idea is that it makes it clear that the returned value borrows data, without all the boilerplate of manually writing out the lifetimes.